src/components/NavBar/index.tsx

```diff
@@ -0,0 +1,108 @@
+'use client';
+
+import { useMemo } from 'react';
+import Link from 'next/link';
+import { usePathname } from 'next/navigation';
+import Icon from '@/components/Icon';
+import COLORS from '@/styles/colors';
+import { Flex } from '@/styles/containers';
+import { useProfile } from '@/utils/ProfileProvider';
+import { ProfileButton, SmallLinkButton } from '../Buttons';
+import * as Styles from './style';
+
+export default function NavBar() {
+  const profile = useProfile();
+  const AuthButtonView = useMemo(() => {
+    if (!profile) throw new Error('Profile must be defined.');
+    if (profile.profileReady && profile?.profileData)
+      return (
+        <ProfileButton href="/settings">
+          {profile.profileData?.first_name || 'Profile'}
+        </ProfileButton>
+      );
```

Comment:
this is mb, the check should instead be `auth.userId` - which means we need to `useAuth` here (note: and the corresponding error check)